mobile/src/main/java/org/openhab/habdroid/ui/MainActivity.java

```diff
@@ -1208,8 +1211,14 @@ public void promptForDevicePassword() {
             return;
         }
         KeyguardManager km = (KeyguardManager) this.getSystemService(Context.KEYGUARD_SERVICE);
-        Intent intent = km.createConfirmDeviceCredentialIntent(getString(R.string.app_name), "");
-        startActivityForResult(intent, SCREEN_LOCK_REQUEST_CODE);
+        if ((Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && km.isDeviceSecure())
+                || (Build.VERSION.SDK_INT < Build.VERSION_CODES.M && km.isKeyguardSecure())) {
+            Intent intent = km.createConfirmDeviceCredentialIntent(getString(R.string.app_name), "");
+            startActivityForResult(intent, SCREEN_LOCK_REQUEST_CODE);
+        } else {
+            showSnackbar(R.string.error_no_pin_set); // TODO
```

Comment:
This function is called, but the Snackbar never shown.
Also: Add a action here and open the Android settings.

Reply:
It's not shown because it's both
- replaced by the 'using local/remote/demo connection' snackbar and
- hidden in onAvailableConnectionChanged